crates/nu-command/src/conversions/into/datetime.rs

```diff
@@ -256,51 +270,59 @@ fn action(input: &Value, args: &Arguments, head: Span) -> Value {
         }
     };
 
-    if let Ok(ts) = timestamp {
-        macro_rules! match_datetime {
-            ($expr:expr) => {
-                match $expr {
-                    dt => Value::Date {
-                        val: dt.into(),
-                        span: head,
-                    },
-                }
-            };
-        }
+    if dateformat.is_none() {
+        if let Ok(ts) = timestamp {
+            macro_rules! match_datetime {
+                ($expr:expr) => {
+                    match $expr {
+                        dt => Value::Date {
+                            val: dt.into(),
+                            span: head,
+                        },
+                    }
+                };
```

Comment:
This doesn't have to be a macro and can be a generic function with a `T: Into<i64>` bound.

Reply:
I'm afraid this is over my pay grade. I played around with it for a while but couldn't figure it out.

Reply:
Ok, I had another thought and hacked something together in d5d0178bc3d187112e68e5ec07a8a9228a1204f9

Reply:
That looks fine as well!